packages/react-server/core/ClientController.js

```diff
@@ -650,7 +650,11 @@ class ClientController extends EventEmitter {
 
 			_.forEach(
 				getRootElementAttributes(element),
-				(v, k) => root.setAttribute(k, v)
+				(v, k) => {
+					if (k !== 'tagName') {
+						root.setAttribute(k, v)
+					}
+				}
```

Comment:
This bit actually isn't necessary.  The attributes we pass through are [white listed in RootElement.js](https://github.com/redfin/react-server/blob/master/packages/react-server/core/components/RootElement.js#L102-L105).